src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -473,6 +473,11 @@ public void CheckSignature(X509Certificate2Collection extraStore, bool verifySig
 
         public void CheckHash()
         {
+            if (_signatureAlgorithm.Value != Oids.NoSignature)
```

Comment:
In Framework, yes (from Win32).

In Core we would accept a "the issuer and serial number look like NoSignature, but the algorithm ID was wrong".  Since NetFX wouldn't produce such a thing, CoreFX couldn't produce any NoSignature, and it's a .NET Framework-ism (not a standard).